packages/fbjs/src/functional/partitionObjectByKey.js

```diff
@@ -18,10 +18,10 @@ var partitionObject = require('partitionObject');
  * `whitelistObjectKeys`, but eventually keeping all the keys. Returns a tuple
  * of objects `[first, second]`.
  */
-function partitionObjectByKey(
-  source: Object,
+function partitionObjectByKey<Tv>(
+  source: {[key: string]: Tv},
   whitelist: Set<string>
-): Array<Object> {
+): [{[key: string]: Tv}, {[key: string]: Tv}] {
```

Comment:
I don't have a nice solution either, but it would work if we change [`partitionObject`](https://github.com/facebook/fbjs/blob/master/packages/fbjs/src/functional/partitionObject.js#L18) to the following and keep `partitionObjectByKey` unchanged:
```javascript
function partitionObject(
  object: Object,
  callback: (value: string, key: string, object: Object) => boolean,
  context?: any
): Array<Object> {
...
}
```